File: dashboard/src/layout/detail.ts
/** 详情面板 —— 右栏 / Focus 模式 */
import { Box, Text } from "@opentui/core";
import { state } from "../state";
import { COLORS, statusIcon } from "../theme";

export function renderDetail(renderer: any, focusMode: boolean): any {
  const focused = state.focusedPanel === "detail";
  const borderColor = focused ? COLORS.borderFocused : COLORS.borderInactive;

  // 找到当前选中的 worker
  const worker = state.workers.find(
    (w) => w.session_id === state.selectedSessionId
  );

  const children: any[] = [
    Text({ content: " Detail ", fg: COLORS.accent, bold: true }),
  ];

  if (worker) {
    const { icon, color } = statusIcon(worker.status);
    children.push(Text({ content: `${icon} ${worker.agent}`, fg: color, bold: true }));
    children.push(Text({ content: worker.model || "?", fg: COLORS.subtext }));

    if (focusMode) {
      const uptime = formatUptime(worker.started_at);
      children.push(
        Text({ content: `${uptime} · ${worker.status.toUpperCase()}`, fg: COLORS.subtext })
      );
      children.push(Text({ content: `Project: ${worker.project}`, fg: COLORS.subtext }));
      children.push(Text({ content: "", fg: COLORS.subtext }));

      // 显示聊天历史（来自 state.messages）
      const msgs = state.messages.get(worker.session_id) || [];
      if (msgs.length > 0) {
        children.push(Text({ content: "▸ Chat", fg: COLORS.accent, bold: true }));
        // 显示最近 10 条
        for (const m of msgs.slice(-10)) {
          const prefix = m.role === "user" ? "你: " : "AI: ";
          const color = m.role === "user" ? COLORS.accent : COLORS.text;
          // 截断长消息
          const lines = m.content.split("\n").slice(0, 4);
          for (let i = 0; i < lines.length; i++) {
            const line = (i === 0 ? prefix : "    ") + lines[i].slice(0, 70);
            children.push(Text({ content: line, fg: color }));
          }
          if (m.streaming) {
            children.push(Text({ content: "▍", fg: COLORS.accent }));
          }
        }
      } else {
        children.push(Text({ content: "▸ Chat (empty)", fg: COLORS.subtext }));
        children.push(Text({ content: "Type a message below and press Enter", fg: COLORS.subtext }));
      }
    } else {
      const log = worker.log_short || "(no output)";
      children.push(Text({ content: "", fg: COLORS.subtext }));
      children.push(Text({ content: `▸ ${log.slice(0, 50)}`, fg: COLORS.text }));
    }
  } else {
    children.push(
      Text({
        content: "Select a worker card\n(Tab to kanban, Enter to select)",
        fg: COLORS.subtext,
      })
    );
  }

  // 日志容器（始终显示，放在详情底部）
  children.push(Text({ content: "", fg: COLORS.subtext }));
  children.push(Text({ content: "▸ Logs", fg: COLORS.accent, bold: true }));
  const recentLogs = state.logs.slice(0, 6);
  if (recentLogs.length === 0) {
    children.push(Text({ content: "(no logs)", fg: COLORS.subtext }));
  } else {
    for (const line of recentLogs) {
      children.push(Text({ content: line, fg: COLORS.subtext }));
    }
  }

  return Box(
    {
      borderStyle: "rounded",
      borderColor,
      bg: COLORS.panelBg,
      padding: 1,
      flexDirection: "column",
      flexGrow: 1,
    },
    children
  );
}

function formatUptime(startedAt?: number): string {
  if (!startedAt) return "--:--:--";
  const now = Date.now();
  const secs = Math.max(0, Math.floor((now - startedAt) / 1000));
  const h = Math.floor(secs / 3600);
  const m = Math.floor((secs % 3600) / 60);
  const s = secs % 60;
  return `${String(h).padStart(2, "0")}:${String(m).padStart(2, "0")}:${String(s).padStart(2, "0")}`;
}

File: src/tui/view/detail.rs
use ratatui::{
    Frame, layout::{Rect, Layout, Constraint, Direction},
    widgets::{Block, Borders, Paragraph, Wrap},
    style::{Style, Modifier},
    text::{Text, Span, Line},
};
use crate::tui::{
    app::{AppState, Panel},
    theme::Theme,
};

pub fn render(f: &mut Frame, state: &mut AppState, area: Rect, theme: Theme, focused: bool) {
    let session_id = match &state.active_session {
        Some(sid) => sid.clone(),
        None => {
            let block = Block::default()
                .title(" Detail ")
                .borders(Borders::ALL)
                .border_style(Style::default().fg(theme.border_inactive))
                .style(Style::default().bg(theme.panel_bg));
            f.render_widget(Paragraph::new("Select a worker card (Enter)").block(block).style(Style::default().fg(theme.subtext)), area);
            return;
        }
    };

    let worker = state.workers.iter().find(|w|
        w.get("session_id").and_then(|v| v.as_str()) == Some(&session_id)
    );

    let border_color = if focused { theme.border_focused } else { theme.border_inactive };

    if state.focus_mode {
        let chunks = Layout::default()
            .direction(Direction::Horizontal)
            .constraints([Constraint::Fill(1), Constraint::Length(18)])
            .split(area);
        let main = chunks[0];
        let side = chunks[1];

        let main_chunks = Layout::default()
            .direction(Direction::Vertical)
            .constraints([Constraint::Fill(1), Constraint::Length(3)])
            .split(main);

        // Chat content
        let agent = worker.and_then(|w| w.get("agent").and_then(|v| v.as_str())).unwrap_or("?");
        let title = format!(" {} · {} ", session_id.chars().take(8).collect::<String>(), agent);

        let content = Paragraph::new(render_chat(state, &session_id))
            .block(Block::default()
                .title(title)
                .borders(Borders::ALL)
                .border_style(Style::default().fg(border_color))
                .style(Style::default().bg(theme.panel_bg)))
            .wrap(Wrap { trim: false });
        f.render_widget(content, main_chunks[0]);

        // Input
        let input_block = Block::default()
            .title(" Input ")
            .borders(Borders::ALL)
            .border_style(Style::default().fg(
                if state.focused_panel == Panel::Input { theme.border_focused } else { theme.border_inactive }
            ))
            .style(Style::default().bg(theme.panel_bg));
        let input_para = Paragraph::new(state.input_text.as_str())
            .block(input_block)
            .style(Style::default().fg(theme.text));
        f.render_widget(input_para, main_chunks[1]);

        // Side panel — simple todo placeholder
        let todo_block = Block::default()
            .title(" Panel ")
            .borders(Borders::ALL)
            .border_style(Style::default().fg(theme.border_inactive))
            .style(Style::default().bg(theme.panel_bg));
        let todo_para = Paragraph::new("Todo / Memory\n(coming soon)")
            .block(todo_block)
            .style(Style::default().fg(theme.subtext));
        f.render_widget(todo_para, side);
    } else {
        // Non-focus: compact detail in right column
        let agent = worker.and_then(|w| w.get("agent").and_then(|v| v.as_str())).unwrap_or("?");
        let model = worker.and_then(|w| w.get("model").and_then(|v| v.as_str())).unwrap_or("?");
        let log = worker.and_then(|w| w.get("log_short").and_then(|v| v.as_str())).unwrap_or("(no output)");
        let lines = vec![
            Line::from(Span::styled(agent, Style::default().fg(theme.accent).add_modifier(Modifier::BOLD))),
            Line::from(Span::styled(model, Style::default().fg(theme.subtext))),
            Line::from(""),
            Line::from(Span::styled(log, Style::default().fg(theme.text))),
        ];
        let p = Paragraph::new(Text::from(lines))
            .block(Block::default()
                .title(" Detail ")
                .borders(Borders::ALL)
                .border_style(Style::default().fg(border_color))
                .style(Style::default().bg(theme.panel_bg)))
            .wrap(Wrap { trim: false });
        f.render_widget(p, area);
    }
}

fn render_chat<'a>(state: &'a AppState, session_id: &str) -> Text<'a> {
    let mut lines: Vec<Line<'a>> = Vec::new();
    if let Some(buf) = state.log_buffers.get(session_id) {
        for l in buf.iter() {
            lines.push(Line::from(Span::raw(l.as_str())));
        }
    }
    if lines.is_empty() {
        lines.push(Line::from(Span::raw("(waiting for output...)")));
    }
    Text::from(lines)
}
